src/components/DisplayToggle/DisplayToggle.component.jsx

```diff
@@ -1,14 +1,24 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import Switch from 'react-switch';
 import { DisplayToggleWrapper } from './DisplayToggle.styled';
+import { useTheme } from '../../providers/Theme';
 
 function DisplayToggle() {
+  const { setTheme } = useTheme();
   const [checked, setChecked] = useState(false);
 
   const toggle = React.useCallback(() => {
     setChecked((v) => !v);
   }, []);
 
+  useEffect(() => {
+    if (checked) {
```

Comment:
I think this useEffect is not necessary. You can add this logic to your toggle function.